### PR TITLE
Fix annotations, bumping pypiserver and cleanup

### PR DESCRIPTION
**Pull request by mekza**
**What this PR does / why we need it**:
- Adding podAnnotations and podLabels
- Removing deployment labels
- Bumping `pypiserver` version
- Adding `apiVersion` to `Chart.yaml`
- Rewording comment for `auth.credentials` map

**Checklist**:
- [x] Bumped chart version in Chart.yaml
- [x] Added an entry in the CHANGELOG.md

### CODE REVIEW

File: pypiserver/values.yaml [hunk 1]
```diff
@@ -18,13 +18,18 @@ auth:
   ## Use '.' or '' for empty. Requires to have set the password (option below).
   ## Available actions are update, download and list
   actions: update
-  ## List of username / encoded passwords that will be put to the htpasswd file
+  ## Map of username / encoded passwords that will be put to the htpasswd file
   ## use `htpasswd -n -b username password` to generate them
   credentials: {}
 
 deployment:
   labels: {}
```
Comment (ClementGautier) [inline pypiserver/values.yaml]: Adding podLabels deprecates usage of deployment.labels, you can remove it.

```suggestion
```

File: pypiserver/templates/deployment.yaml [hunk 2]
```diff
@@ -21,8 +21,15 @@ spec:
       labels:
         app.kubernetes.io/name: {{ template "pypiserver.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
-        {{- if .Values.deployment.labels }}
-        {{- toYaml .Values.deployment.labels | nindent 8 }}
+        {{- if .Values.podLabels }}
+        {{- toYaml .Values.podLabels | nindent 8 }}
+        {{- end }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
```
Comment (ClementGautier) [inline pypiserver/templates/deployment.yaml]: `secret` makes more sense to me (so if we introduce a configmap at some point we could still use checksum/config for that file.)
```suggestion
        checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
```

File: pypiserver/Chart.yaml [hunk 3]
```diff
@@ -1,7 +1,8 @@
+apiVersion: v1
 name: pypiserver
 home: https://github.com/pypiserver/pypiserver
-version: 1.1.0
-appVersion: 1.3.0
+version: 1.2.0
```
Comment (ClementGautier) [inline pypiserver/Chart.yaml]: podLabels replaces deployment.labels, it's a breaking change so I'd be more confortable bumping a major here

```suggestion
version: 2.0.0
```

File: pypiserver/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,18 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [1.2.0]
```
Comment (ClementGautier) [inline pypiserver/CHANGELOG.md]:
```suggestion
## [2.0.0]

### Breacking Changes

- `podLabels` replaces `deployment.labels`
```

**Comment by ClementGautier** [comment]
I made a last commit on the branch to complete the README and merged your commit in another PR. Thanks a lot for your contribution!

**Comment by mekza** [comment]
@ClementGautier You are welcome! You could have checkout my branch and make the changes directly but I am happy you merged it.

**Pull request closed without merging**